cmp/report_compare.go

```diff
@@ -81,14 +81,20 @@ func (opts formatOptions) FormatDiff(v *valueNode) textNode {
 		return opts.FormatDiffSlice(v)
 	}
 
+	var isSlice bool
```

Comment:
Perhaps call this `withinSlice` instead (here and elsewhere)? Since we're usually looking at this variable from the perspective of the child.